### DISCUSSION TITLE
Watch vs useWatch

### DISCUSSION THREAD

**Discussion by JasonTheAdams** [react-hook-form/react-hook-form#4455] [Q&A] [answered]
Greetings!

I'm trying to understand the differences between `watch` and `useWatch`, specifically in v7. They both provide very similar purposes. It sounds like, in terms of rendering, `useWatch` is far better as `watch` will re-render the entire "app or form" (from the docs, when is it one or the other?).

Is the simple difference that `watch` is a function and can be used anywhere? Whereas `useWatch` is a hook and therefore must be used within a component or hook? It seems like `useWatch` should be preferred over `watch`, being used whenever possible over the other. Is this correct?

Thanks for all the amazing work!

**Comment by bluebill1049** [comment, marked as answer]
hey @JasonTheAdams watch re-render at root, useWatch re-render at the component level.

**Reply by JasonTheAdams** [reply]
Hey Bill!

Got it. So they're functionally identical. The real question is _what_ you want to have re-rendered in the case that what's being watched does change. Correct?

**Reply by bluebill1049** [reply]
That's correct @JasonTheAdams they use the same internal logic, it's just where gets re-rendered.

**Reply by JasonTheAdams** [reply]
Roger that. Thank you, Bill! 🙌

**Reply by Aryaman73** [reply]
@bluebill1049 does this mean that we should be using `useWatch` wherever possible?

**Reply by Aryaman73** [reply]
Also, how does `getValues` fit into this?

**Reply by bluebill1049** [reply]
`watch` - render the entire form (simple and easy) works well with a simple form 
`useWatch` - isolate render in a component level (more work) works well with large and complex form
`getValues` - only return form values once, no subscription.

hope above make sense.
